invenio_communities/records/records/systemfields/communities/manager.py

```diff
@@ -0,0 +1,205 @@
+        self._communities_ids.add(community_id)
+
+        # Set default
+        if default:
+            self._default_id = community_id
+
+        # Cache community only if provided
+        if isinstance(community_or_id, Community):
+            self._communities_cache[community_id] = community_or_id
+
+    def remove(self, community_or_id):
+        """Remove a record from a community."""
+        community_id = self._to_id(community_or_id)
+
+        # Delete M2M row.
+        res = self._m2m_model_cls.query.filter_by(
+            community_id=community_id,
+            record_id=self._record_id
+        ).delete()
+        if res != 1:
+            raise ValueError('The record has not been added to the community.')
+
+        # Remove from internal set
+        self._communities_ids.remove(community_id)
+
+        # Unset default if needed
+        if self._default_id == community_id:
+            self._default_id = None
+
+    def clear(self):
+        """Clear all communities from the record."""
+        # Remove all associations
+        res = self._m2m_model_cls.query.filter_by(
+            record_id=self._record_id).delete()
+        self._communities_ids = set()
+        self._default_id = None
+        self._communities_cache = {}
+
+    def refresh(self):
+        """Refresh from the database M2M table."""
+        # Retrieve from M2M table
+        ids = db.session.query(
+            self._m2m_model_cls.community_id
+        ).filter(
+            self._m2m_model_cls.record_id == self._record_id
+        ).all()
+
+        # Set internal list
+        self._communities_ids = set([str(x[0]) for x in ids])
+
+        # Unset default if no longer available
+        if self._default_id and self._default_id not in self._communities_ids:
+            self._default_id = None
+
+    def __len__(self):
+        """Get number of communities"""
+        return len(self._communities_ids)
+
+    def __contains__(self, community_or_id):
+        """Check record is in community."""
+        id_ = self._to_id(community_or_id)
+        return id_ in self._communities_ids
+
+    def __iter__(self):
+        """Iterate over a communities."""
+        # Determine community ids not already cached.
+        nocache_ids = self._communities_ids - set(
+            self._communities_cache.keys())
+
+        # Fetch and cache missing community records
+        if nocache_ids:
+            communities = Community.get_records(nocache_ids)
+            for c in communities:
+                self._communities_cache[str(c.id)] = c
+
+        # Iterate (sort by identifier to ensure consistent results)
+        return (
+            self._communities_cache[c]
+            for c in sorted(self._communities_ids)
+        )
+
+    @property
+    def ids(self):
+        """Get communities ids."""
+        return sorted(self._communities_ids)
+
+    @property
+    def default(self):
+        """Get the default community."""
+        if self._default_id:
```

Comment:
could it happen here too, that the `_default_id` is something falsy but different from `None`?
e.g. `id == 0` (unlikely b/c it's usually UUIDs, but still)?

Reply:
Yep, better safe than sorry - don't want to spend another hour debugging :D